server/debug.go

```diff
@@ -29,6 +29,37 @@ func (s *Server) registerDebugRoutes(g *echo.Group) {
 
 		return currentDebugState(c)
 	})
+
+	g.GET("/debug/log", func(c echo.Context) error {
+		var errorRecordList []*api.DebugLog
+		var incrID int
+
+		s.errorRecordRing.Mutex.RLock()
+		defer s.errorRecordRing.Mutex.RUnlock()
+
+		s.errorRecordRing.Ring.Do(func(p interface{}) {
+			if p != nil {
```

Comment:
Could this be nil?

Reply:
This check is necessary since `ring.Do` will consistently execute 100 times even if the ring is not full.
If the current value is empty, we should skip it.

Reply:
Used the following code instead.

Less indentation and id should start with zero as previous comment stated.

		s.errorRecordRing.Ring.Do(func(p interface{}) {
			if p == nil {
				return
			}
			errRecord, ok := p.(*api.ErrorRecord)
			if !ok {
				return
			}
			errorRecordList = append(errorRecordList, &api.DebugLog{
				ID:          incrementID,
				ErrorRecord: *errRecord,
			})
			incrementID++
		})